sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -1290,6 +1291,7 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
         res.filter(p => isPartialPartitionSpec(spec, toMetaStorePartitionSpec(p.spec)))
       case _ => res
     }
+    parts.map(restorePartitionMetadata(_, catalogTable))
```

Comment:
Actaully, `getPartition` does `restorePartitionMetadata` as expected. But `listPartitions` doesn't follow this behavior.

Reply:
BTW, `restorePartitionMetadata ` is added by SPARK-21213. From its description, seems at the moment, the patch proposed to analyze table partition to collect partition-level statistics.

Reply:
So I guess it was not considered for the case of listing partitions.

Reply:
Yes looks like not all operations are updated with `restorePartitionMetadata`